### PR TITLE
CM-APP-42 | API changes for fetching list of consent artefacts (with: status, limit & offset)

### PR DESCRIPTION
**Pull request by shwetadongare**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactsController.java  1
- src/main/java/in/projecteka/consentmanager/consent/model/ListResult.java  1
         

Clones removed
==============
+ src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5507512&bid=18049841)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactsController.java  1
- src/main/java/in/projecteka/consentmanager/consent/model/ListResult.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  1
         

Clones removed
==============
+ src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5507512&bid=18049841)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/model/ListResult.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactsController.java  1
         

Clones removed
==============
+ src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5507512&bid=18049841)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactsController.java  1
- src/main/java/in/projecteka/consentmanager/consent/model/ListResult.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  1
         

Clones removed
==============
+ src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5507512&bid=18049841)

File: src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java [hunk 1]
```diff
@@ -135,6 +146,56 @@
                         }));
     }
 
+    public Mono<ListResult<List<ConsentArtefactRepresentation>>> getAllConsentArtefacts(
+            String username, int limit, int offset) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(SELECT_ALL_CONSENT_ARTEFACTS)
+                .execute(Tuple.of(username, limit, offset), handler -> {
+                    List<ConsentArtefactRepresentation> artefacts = getConsentArtefactRepresentation(monoSink, handler);
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java:153]: Why pass the monoSink as argument to the method. Handle all success or error here, based on result.

File: src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java [hunk 2]
```diff
@@ -27,30 +32,44 @@
...
             "FROM consent_artefact WHERE status=$1";
     private static final String UPDATE_CONSENT_ARTEFACT_STATUS_QUERY = "UPDATE consent_artefact SET status=$1, " +
             "date_modified=$2 WHERE consent_artefact_id=$3";
+    private static final String SELECT_CONSENT_ARTEFACTS_COUNT_FOR_STATUS = "SELECT COUNT(*) FROM consent_artefact " +
+            "WHERE patient_id=$1 AND status=$2";
+    private static final String SELECT_CONSENT_ARTEFACTS_COUNT = "SELECT COUNT(*) FROM consent_artefact " +
+            "WHERE patient_id=$1";
+
+    static {
+        String s = "SELECT status, consent_artefact, signature FROM ";
```
[the remark this answers is not in the document]
Comment (shwetadongare) [reply src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java:62]: There is no specific advantage, but sonar pointed it out as a code smell as the select part of it was same at 4 places.

File: src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java [hunk 3]
```diff
@@ -135,6 +146,56 @@
                         }));
     }
 
+    public Mono<ListResult<List<ConsentArtefactRepresentation>>> getAllConsentArtefacts(
+            String username, int limit, int offset) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(SELECT_ALL_CONSENT_ARTEFACTS)
+                .execute(Tuple.of(username, limit, offset), handler -> {
+                    List<ConsentArtefactRepresentation> artefacts = getConsentArtefactRepresentation(monoSink, handler);
+                    if (artefacts == null) return;
```
[the remark this answers is not in the document]
Comment (shwetadongare) [reply src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java:154]: Actually that is already handled. The if is redundant, will remove that.

File: src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java (hunk 1 above)
Comment (shwetadongare) [reply src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java:153]: Agreed,  will make that change.

**Pull request merged**